### PR TITLE
tweak editor

### PR DESCRIPTION
**Pull request by tophf**
* hide focus outline on a lone section

WDYT?

* also apply live-preview if an unsaved style was disabled - useful in case live-preview was disabled at first, then after some editing you uncheck `[ ] enabled` (or press Alt-Enter), and enable live-preview to see how the site looks without the style applied.

### CODE REVIEW

**Comment by narcolepticinsomniac** [comment]
> hide focus outline on a lone section

I prefer the visual indicator, lone section or not.

> use standard CSS for a focused CodeMirror outline

Better, but still odd looking. Edges are uneven, and corners are wonky. They're doing some stupid stuff with outlines now. It's like they're partly inset now by default, which appears uneven on certain elements, depending on their properties.

Since editors are so prominent, let's just get rid of outlines and use box-shadow instead, since it's visual appearance is more reliable. I'd say we switch it to a little more of a cyan shade, since it's on brand. We can tweak the styling, but something like:

```
.CodeMirror {
  border: solid #CCC 1px;
  outline: none;
  transition: box-shadow .2s;
}
.CodeMirror-lint-mark-warning {
  background: none;
}
.CodeMirror-dialog {
  -webkit-animation: highlight 3s cubic-bezier(.18, .02, 0, .94);
}
.CodeMirror-focused {
  box-shadow: 0 0 0 1px hsl(180, 100%, 38%), 0 0 3px hsla(180, 100%, 60%, .5); /* not using outline as it became ugly in new Chrome */
}
```

**Comment by narcolepticinsomniac** [comment]
Testing TAB accessibility, and section CM editors kinda seem like traps. Wasn't there some trick to reverse, and tab through them?

**Comment by tophf** [comment]
The box-shadow looks good. We should probably use the same style for the keyboard focus outline instead of the built-in one atrocity.

>I prefer the visual indicator, lone section or not.

My idea was that it's probably weird to have it on a lone section as we explicitly hide the focus outline everywhere except when the keyboard is used to navigate.

>Testing TAB accessibility

Use Ctrl-Break.

**Comment by narcolepticinsomniac** [comment]
>use box-shadow instead of outline for focus everywhere

Yeah, to hell with their fugly outlines. =)

>it's probably weird to have it on a lone section as we explicitly hide the focus outline everywhere

We might've gone a little overboard suppressing focus styling on inputs. Probably a good idea to give em the box-shadow and quick transition, like editors. Focus styling is super annoying on click-action elements, like buttons, checkboxes/labels, icons, and links, but I find it useful to have some subtle indication for active inputs, and textareas.

It's helpful being able to instantly tell if an editor's focused when switching back to a tab. As it is, my dumb ass occasionally starts coding away into the ether.

**Comment by narcolepticinsomniac** [comment]
>Use Ctrl-Break

No Break button here. I looked it up, and they suggest alternatives for different manufacturers, but none seem to work. I don't care much about accessibility for my own personal use, but we've gone through a lot of trouble making it decent for users, so I hope that anyone who actually needs it doesn't run into the same issue.

**Comment by tophf** [comment]
Okay, sounds convincing so I've implemented that. And while the other PRs are blocked on eight's timeout, let's treat this one as a "dev channel" so I'm adding a few fixes for USO and cleanups of no longer necessary Chrome 49 workarounds.

>No Break button here.
Another name of the Break key is `Pause` so maybe you have that one.

**Comment by narcolepticinsomniac** [comment]
Hey, I hooked up some focus styling improvements. You don't publish your branches when you open PRs, so I'm not sure how to commit here, and I don't feel like guessing wrong. LMK what you'd prefer.

**Comment by tophf** [comment]
All my PRs allow write access by default to openstyles members so you can checkout the branch locally and push into it. I don't know if you can do it on github.com site though. BTW I hope you didn't add transition there as I'll fight ya on that, I can't stand them.

**Comment by narcolepticinsomniac** [comment]
I literally have no idea what happened the first time. =) Shaved transitions down to .1s, so hopefully that's a compromise you can live with. Mostly, it fixes some inputs we were missing, and a couple styling tweaks

File: global.css [hunk 1]
```diff
@@ -210,11 +211,14 @@ select[disabled] + .select-arrow {
 }
 
 :focus,
-.CodeMirror-focused {
+.CodeMirror-focused,
+[data-focused-via-click] input[type="text"]:focus,
+[data-focused-via-click] input[type="number"]:focus {
```
Comment (tophf) [inline global.css:216]: just in case, `search` isn't needed here?

Comment (narcolepticinsomniac) [reply global.css:216]: These are for usercss config, and option labels, so I don't think so. Couldn't hurt though.

File: js/dom.js [hunk 2]
```diff
@@ -357,7 +357,8 @@ function focusAccessibility() {
   const isOutlineAllowed = ({localName, type}) =>
     !focusAccessibility.ELEMENTS.includes(localName) ||
     // allow outline on text/search inputs in addition to textareas
-    localName === 'input' && (type === 'text' || type === 'search');
+    localName === 'input' &&
+    !/^(button|checkbox|color|file|image|radio|submit|range)$/.test(type);
```
Comment (tophf) [inline js/dom.js:361]: I would use a positive check for `/^(text|search|number)$/` because it's shorter

**Comment by tophf** [comment]
> Shaved transitions down to .1s, so hopefully that's a compromise you can live with

I'll try for a while. If I can't, I would probably just add a global `@media (prefers-reduced-motion) { * { animation: none !important } }`.

File: js/dom.js (hunk 2 above)
Comment (narcolepticinsomniac) [reply js/dom.js:361]: Well, it's future proof, plus some text inputs don't have type attributes.

Comment (tophf) [reply js/dom.js:361]: `type` is never empty inside JS so it'll be `text`. As for the future, I could argue a fancy control is more likely to be added whereas the classic ones for text/search/number are more likely to keep going for ages.

Comment (narcolepticinsomniac) [reply js/dom.js:361]: >`type` is never empty inside JS so it'll be `text`.

News to me. Some weren't working, but it must've just been the data attribute on a parent. Positive check should be fine then.

Comment (tophf) [reply js/dom.js:361]: Yeah by "JS" I mean direct access to the element's property like we do here, not via getAttribute or dataset and such.

**Comment by narcolepticinsomniac** [comment]
WTH did I do now? How did editor-focus branch become involved? Man, I wish you'd just publish normally. I'm too simple for this.

**Comment by tophf** [comment]
Nothing bad happened. I've just pushed a commit into this PR after yours. When you pushed your new commit, github automatically added mine into your local branch. For universal reliability it merges of all the new stuff that your local branch didn't include as one merge commit. It won't matter when we squash the PR into the master branch anyway.

**Comment by narcolepticinsomniac** [comment]
Right on. I totally know what I'm doing. =)

**Comment by tophf** [comment]
Merging this one as it's simple enough and I need it in the custom name PR to improve on the editor load speed even further.

**Comment by tophf** [comment]
#1054 is the bleeding edge now, let's move there to test and tweak it.

**Pull request merged**